### ISSUE TITLE
Bug: time-limit bootstrap ignores final_observation in synchronous PPO, and APPO/off-policy workers reuse patched terminal obs as next actor input

### ISSUE THREAD

**Issue by TATP-233** [Motphys/UniLab#236]
### Summary

`terminated` and `truncated` are split in the common env contract, but the current training pipelines do not consume that contract strictly.

Two problems are visible from repository code and tests:

1. Synchronous PPO paths do not bootstrap time-limit transitions from `info["final_observation"]`; they use the current-step value while `NpEnv.step()` has already reset done envs and replaced `state.obs` with the next episode's reset observation.
2. APPO and off-policy workers patch `final_observation` into `next_obs` for storage, but then also reuse that patched tensor as the next actor input, which can desynchronize the policy input from the env's internal post-reset state.

### Area

algos

### Reproduction

This can be reproduced from the repository contract and targeted code inspection without relying on benchmark outcomes.

1. Confirm the env contract and `final_observation` behavior:

```bash
uv run pytest tests/base/test_np_env.py -k final_observation
```

Relevant evidence:

- `src/unilab/base/np_env.py` stores pre-reset terminal obs into `info["final_observation"]` and then resets done envs in-place.
- `tests/base/test_np_env.py` verifies that `state.obs` for done envs is already reset obs, while `info["final_observation"]` preserves the pre-reset terminal obs.

2. Inspect synchronous PPO paths:

```bash
sed -n '94,109p' src/unilab/utils/rsl_rl_vec_env_wrapper.py
sed -n '458,467p' scripts/train_mlx_ppo.py
```

Observed code path:

- `RslRlVecEnvWrapper.step()` exports `state.done` and `state.truncated` as `infos["time_outs"]`, but does not forward `final_observation`.
- `scripts/train_mlx_ppo.py` applies timeout correction with `rewards += gamma * values_mx * timeouts`, but also does not read `final_observation`.

3. Inspect APPO and off-policy workers:

```bash
sed -n '178,197p' src/unilab/algos/torch/appo/worker.py
sed -n '262,318p' src/unilab/algos/torch/offpolicy/worker.py
```

Observed code path:

- Both workers patch `state.info["final_observation"]` into the transition's next observation.
- Both then continue with that patched observation as the next loop iteration's actor input (`obs_np = next_obs_np` or equivalent), even though `NpEnv.step()` has already reset the env.

4. Inspect APPO learner timeout correction:

```bash
sed -n '258,260p' src/unilab/algos/torch/appo/learner.py
```

Observed code path:

- APPO adds `gamma * values * truncated` using the current state's value, not an explicit value of the final pre-reset observation.

### Expected behavior

- For time-limit truncation, bootstrap should be computed from the true final pre-reset observation carried in `info["final_observation"]`, not from `V(obs_t)` and not from the reset observation of the next episode.
- Patching `final_observation` into a stored transition's `next_obs` is acceptable for learning targets, but that patched terminal observation should not become the next actor input after the env has already auto-reset.
- `terminated` should stop bootstrap; `truncated` should preserve bootstrap, but the bootstrap source should remain aligned with the env contract.

### Actual behavior

- In synchronous PPO:
  - `NpEnv.step()` resets done envs immediately and exposes the true terminal observation only via `info["final_observation"]` (`src/unilab/base/np_env.py`).
  - `src/unilab/utils/rsl_rl_vec_env_wrapper.py` forwards `state.truncated` as `infos["time_outs"]` but drops `final_observation`.
  - `scripts/train_mlx_ppo.py` applies timeout correction directly from `values_mx` and `state.truncated`, also without consuming `final_observation`.

- In APPO and off-policy workers:
  - `src/unilab/algos/torch/appo/worker.py` and `src/unilab/algos/torch/offpolicy/worker.py` overwrite `next_obs` with `final_observation` for done envs.
  - The patched observation is then reused as the next actor input, which no longer matches the env's internal post-reset state.

- In APPO specifically:
  - `src/unilab/algos/torch/appo/learner.py` performs timeout correction as `rewards += self.gamma * values.detach() * truncated`, which appears to use `V(s_t)` rather than a value computed on the true final observation.

### Environment

OS: macOS Darwin arm64
Python: 3.13.9 via `uv`
Backend: backend-agnostic issue in common env/algorithm layers; affects MuJoCo and Motrix owners that use these stacks
Device: not hardware-specific from current evidence
Commit: `0134e2a0dd42e3076ce7746019910e030b62e8d0`

### Logs and artifacts

Repository evidence:

- `src/unilab/base/np_env.py`
- `tests/base/test_np_env.py`
- `src/unilab/utils/rsl_rl_vec_env_wrapper.py`
- `scripts/train_mlx_ppo.py`
- `src/unilab/algos/torch/appo/worker.py`
- `src/unilab/algos/torch/appo/learner.py`
- `src/unilab/algos/torch/offpolicy/worker.py`

Supporting artifact already written in this repo:

- `agents/terminated_truncated_audit.md`
